### PR TITLE
Documentation on Optional Services

### PR DESCRIPTION
**Pull request by brittainhard**
Moved info about nutch to "Optional Services" for now, since we can't install it by default.

### CODE REVIEW

**Comment by brittainhard** [comment]
Ready for review @ahmadia

File: docs/source/crawler_guide.rst [hunk 1]
```diff
@@ -54,9 +54,9 @@ Start Button
 
 Stop Button
 -----------
-    Symbolized by the "stop" button. Stops the crawl.
+   Symbolized by the "stop" button. Stops the crawl.
 
-    In the case of Ache, the crawler stops immediately. In the case of Nutch, the crawler stops after it has finished the current round. This is in order to prevent data corruption that can occur when killing the Nutch process.
+   In the case of Ache, the crawler stops immediately. In the case of Nutch, the crawler stops after it has finished the current round. This is in order to prevent data corruption that can occur when killing the Nutch process.
```
Comment (ahmadia) [inline docs/source/crawler_guide.rst]: No longer true.  Now it stops after the current task.  All data from the current crawl round is lost though.

File: docs/source/crawler_guide.rst [hunk 2]
```diff
@@ -34,15 +34,15 @@ Creating a Seeds List
 
 Crawler Control Buttons
 =======================
-Here's an overview of the buttons available to each crawler for controlling the crawlers. The buttons behave differently depending on which crawler you are using.
+   Here's an overview of the buttons available to each crawler for controlling the crawlers. The buttons behave differently depending on which crawler you are using.
```
Comment (ahmadia) [inline docs/source/crawler_guide.rst]: Can we get rid of the redundant "crawlers" here?

File: docs/source/dev_guide.rst [hunk 3]
```diff
@@ -6,9 +6,9 @@ Developer's Guide to Memex Explorer
 Setting up Memex Explorer
 *************************
 
-To setup your machine, you will need Anaconda or Miniconda installed. Miniconda is a minimal Anaconda installation that bootstraps conda and Python on any operating system. Install `Anaconda <http://continuum.io/downloads>`_ or `Miniconda <http://conda.pydata.org/miniconda.html>`_ from their respective sites.
+   To setup your machine, you will need Anaconda or Miniconda installed. Miniconda is a minimal Anaconda installation that bootstraps conda and Python on any operating system. Install `Anaconda <http://continuum.io/downloads>`_ or `Miniconda <http://conda.pydata.org/miniconda.html>`_ from their respective sites.
```
Comment (ahmadia) [inline docs/source/dev_guide.rst]: I still think it's "set up" in verb form :)

File: docs/source/dev_guide.rst [hunk 4]
```diff
@@ -30,12 +30,35 @@ Application Setup
...
 
-Enabling Nutch Visualizations
+Tests
+=====
+    To run the tests, return to the root directory and run:
+
+    .. code-block:: html
+
+       $ py.test
+
```
Comment (ahmadia) [inline docs/source/dev_guide.rst]: Sweet!

File: docs/source/dev_guide.rst [hunk 5]
```diff
@@ -30,12 +30,35 @@ Application Setup
...
+The Database Model
+==================
+   The current entity relation diagram:
+
+.. image:: _static/img/DbVisualizer.png
```
Comment (ahmadia) [inline docs/source/dev_guide.rst]: Has this been committed?

File: docs/source/dev_guide.rst [hunk 6]
```diff
@@ -30,12 +30,35 @@ Application Setup
...
+
+Updating the Database
+---------------------
+   As of version 0.4.0, Memex Explorer will start tracking all database migrations. This means that you will be able to upgrade your database and preserve the data without any issues.
+
+   If you are using a version that is 0.3.0 or earlier, and you are unable to update your database without server errors, the best course if action is to delete the existing `file at source/db.sqlite3` and start over with a fresh database.
```
Comment (ahmadia) [inline docs/source/dev_guide.rst]: course of action.

File: docs/source/dev_guide.rst [hunk 7]
```diff
@@ -30,12 +30,35 @@ Application Setup
...
+
+Enabling Non-Default Services
 =============================
 
+Nutch Visualizations
+--------------------
+
    Nutch visualizations are not enabled by default. Nutch visualizations require RabbitMQ, and the method for installing RabbitMQ varies depending on the operating system. RabbitMQ can be installed via Homebrew on Mac, and apt-get on Debian systems. More information on how to install RabbitMQ, read `this page <https://www.rabbitmq.com/download.html>`_.  Note: You may also need to change the below command to `sudo rabbitmq-server`, depending on how RabbitMQ is installed on your system and the permissions of the current user.
```
Comment (ahmadia) [inline docs/source/dev_guide.rst]: For more information ... ?

File: docs/source/dev_guide.rst [hunk 8]
```diff
@@ -30,12 +30,35 @@ Application Setup
...
 
-   To enable Bokeh visualizations for Nutch, change ``autostart=false`` to ``autostart=true`` for both of these directives in source/supervisord.conf, and then kill and restart supervisor.
+   To enable Bokeh visualizations for Nutch, change ``autostart=false`` to ``autostart=true`` for both of these directives in `source/supervisord.conf`, and then kill and restart supervisor.
```
Comment (ahmadia) [inline docs/source/dev_guide.rst]: should we add a little more contextual information for the bokeh-server and rabbitmq?

**Comment by ahmadia** [comment]
Fantastic work Brittain!  Just a few minor fixes.  Sorry I didn't get to this earlier.

**Comment by brittainhard** [comment]
Merging soon.

**Comment by ahmadia** [comment]
great, thanks!

**Comment by ahmadia** [comment]
Awesome!  +1 to merge.

**Pull request merged**